src/mpi-base.jl

```diff
@@ -156,6 +120,54 @@ function Comm_size(comm::Comm)
     Int(size[])
 end
 
+function Type_create_struct{T <: Any}(::Type{T})  # <: Any effectively 
```

Comment:
Re T being dynamic, I though the `::Type{T}` construct determined the value of `T` at compile time, so the compiler will be able to optimize.  Is that not correct?

Re the function name, yes, a new name would be better.